counsel.el

```diff
@@ -5216,10 +5216,13 @@ filtered out."
 
 ;;;###autoload
 (defun counsel-compile (&optional arg)
-  "Call `compile' completing from compile history and additional suggestions."
-  (interactive)
+  "Call `compile' completing from compile history and additional suggestions.
```

Comment:
How about two dedicated functions as well?

Reply:
Instead of accepting a prefix I've allowed counsel-compile to be called with an explicit dir and then recurse if we go down into a build dir (although previous history is still visible on the first invocation). I'm open to better ways to handle the recursion, I wasn't quite sure how ivy's recursive support was meant to be used.